src/utils/promote.ts

```diff
@@ -299,29 +299,36 @@ async function getOrSelectApplication(
 		$expand: {
 			is_for__device_type: { $select: ['slug', 'is_of__cpu_architecture'] },
 		},
-	} satisfies BalenaSdk.PineOptions<BalenaSdk.Application>;
+	} as const;
 
 	// Check for a fleet slug of the form `user/fleet` and update the API query.
 	let name: string;
 	const match = appName.split('/');
 	if (match.length > 1) {
 		// These will match at most one fleet
-		(options as BalenaSdk.PineOptions<BalenaSdk.Application>).$filter = {
+		(
+			options as BalenaSdk.Pine.ODataOptionsWithoutCount<
```

Comment:
If you are up to if, you can get rid of these casts completely if you define the opts as:
```
	const options = {
		$expand: {
			is_for__device_type: { $select: ['slug', 'is_of__cpu_architecture'] },
		},
		$filter: {}
	} satisfies BalenaSdk.Pine.ODataOptionsWithoutCount<
		BalenaSdk.Application['Read']
	>;
```